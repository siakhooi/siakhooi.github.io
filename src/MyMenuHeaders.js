import React, { useState } from 'react'
import PropTypes from 'prop-types'
import MyMenuResponsiveNav from './MyMenuResponsiveNav'
import MyMenuNav from './MyMenuNav'
import MyMenuHeader from './MyMenuHeader'

MyMenuHeaders.propTypes = {
  displayTab: PropTypes.number,
  o: PropTypes.array,
  setDisplayTab: PropTypes.func
}

function MyMenuHeaders (props) {
  const [displayNav, setDisplayNav] = useState(0)

  let title
  const headers = props.o.map((x, i) => {
    let selected = 'N'
    if (i === props.displayTab) {
      title = x
      selected = 'Y'
    }
    return (
      <MyMenuHeader key={i} id={i} name={x} selected={selected} onClick={() => { props.setDisplayTab(i); setDisplayNav(0) }}></MyMenuHeader>
    )
  })
  return (
    <div className="MyMenuHeaders">
      <MyMenuResponsiveNav title={title} onClick={() => setDisplayNav((displayNav === 1 ? 0 : 1))} />
      <MyMenuNav displayNav={displayNav === 1 ? 'Y' : 'N'}>{headers}</MyMenuNav>
    </div>
  )
}

export default MyMenuHeaders
