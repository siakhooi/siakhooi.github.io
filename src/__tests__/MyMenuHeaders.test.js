import React from 'react'
import renderer from 'react-test-renderer'
import MyMenuHeaders from '../MyMenuHeaders'

it('MyMenuHeaders/1', () => {
  const dummyFunction = jest.fn()

  const headers = ['Header 1', 'Header 2', 'Header 3']

  const component =
    renderer.create(
      <MyMenuHeaders displayTab={0} o={headers} setDisplayTab={dummyFunction} />
    )
  expect(component.toJSON()).toMatchSnapshot()
})

it('MyMenuHeaders/2', () => {
  const dummyFunction = jest.fn()

  const headers = ['Header 1', 'Header 2', 'Header 3']

  let component
  renderer.act(() => {
    component = renderer.create(
      <MyMenuHeaders displayTab={1} o={headers} setDisplayTab={dummyFunction} />
    )
  })
  expect(component.toJSON()).toMatchSnapshot()

  renderer.act(() => {
    component.toJSON()
      .children.find(x => { return x.props.className === 'MyMenuResponsiveNav' })
      .children.find(x => { return x.props.className.indexOf('MyMenuBars') !== -1 })
      .props.onClick() // setDisplayNav
  })
  expect(component.toJSON()).toMatchSnapshot()

  renderer.act(() => {
    component.toJSON()
      .children.find(x => { return x.props.className === 'MyMenuResponsiveNav' })
      .children.find(x => { return x.props.className.indexOf('MyMenuBars') !== -1 })
      .props.onClick()
  })
  expect(component.toJSON()).toMatchSnapshot()

  renderer.act(() => {
    component.toJSON()
      .children.find(x => { return x.props.className === 'MyMenuNav' })
      .children[1]
      .props.onClick() // setDisplayTab
  })
  expect(dummyFunction).toBeCalled()
  expect(component.toJSON()).toMatchSnapshot()

  renderer.act(() => {
    component.toJSON()
      .children.find(x => { return x.props.className === 'MyMenuNav' })
      .children[0]
      .props.onClick()
  })
  expect(component.toJSON()).toMatchSnapshot()
})
