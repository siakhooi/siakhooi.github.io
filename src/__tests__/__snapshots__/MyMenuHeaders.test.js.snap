// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMenuHeaders/1 1`] = `
<div
  className="MyMenuHeaders"
>
  <div
    className="MyMenuResponsiveNav"
  >
    <span
      className="MyMenuTitle"
    >
      Header 1
    </span>
    <button
      className="MyMenuBars fa fa-bars"
      onClick={[Function]}
    />
  </div>
  <div
    className="MyMenuNav"
    displayNav="N"
  >
    <button
      className="MyMenuHeader"
      currentTab="Y"
      onClick={[Function]}
    >
      Header 1
    </button>
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 2
    </button>
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 3
    </button>
  </div>
</div>
`;

exports[`MyMenuHeaders/2 1`] = `
<div
  className="MyMenuHeaders"
>
  <div
    className="MyMenuResponsiveNav"
  >
    <span
      className="MyMenuTitle"
    >
      Header 2
    </span>
    <button
      className="MyMenuBars fa fa-bars"
      onClick={[Function]}
    />
  </div>
  <div
    className="MyMenuNav"
    displayNav="N"
  >
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 1
    </button>
    <button
      className="MyMenuHeader"
      currentTab="Y"
      onClick={[Function]}
    >
      Header 2
    </button>
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 3
    </button>
  </div>
</div>
`;

exports[`MyMenuHeaders/2 2`] = `
<div
  className="MyMenuHeaders"
>
  <div
    className="MyMenuResponsiveNav"
  >
    <span
      className="MyMenuTitle"
    >
      Header 2
    </span>
    <button
      className="MyMenuBars fa fa-bars"
      onClick={[Function]}
    />
  </div>
  <div
    className="MyMenuNav"
    displayNav="Y"
  >
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 1
    </button>
    <button
      className="MyMenuHeader"
      currentTab="Y"
      onClick={[Function]}
    >
      Header 2
    </button>
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 3
    </button>
  </div>
</div>
`;

exports[`MyMenuHeaders/2 3`] = `
<div
  className="MyMenuHeaders"
>
  <div
    className="MyMenuResponsiveNav"
  >
    <span
      className="MyMenuTitle"
    >
      Header 2
    </span>
    <button
      className="MyMenuBars fa fa-bars"
      onClick={[Function]}
    />
  </div>
  <div
    className="MyMenuNav"
    displayNav="N"
  >
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 1
    </button>
    <button
      className="MyMenuHeader"
      currentTab="Y"
      onClick={[Function]}
    >
      Header 2
    </button>
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 3
    </button>
  </div>
</div>
`;

exports[`MyMenuHeaders/2 4`] = `
<div
  className="MyMenuHeaders"
>
  <div
    className="MyMenuResponsiveNav"
  >
    <span
      className="MyMenuTitle"
    >
      Header 2
    </span>
    <button
      className="MyMenuBars fa fa-bars"
      onClick={[Function]}
    />
  </div>
  <div
    className="MyMenuNav"
    displayNav="N"
  >
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 1
    </button>
    <button
      className="MyMenuHeader"
      currentTab="Y"
      onClick={[Function]}
    >
      Header 2
    </button>
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 3
    </button>
  </div>
</div>
`;

exports[`MyMenuHeaders/2 5`] = `
<div
  className="MyMenuHeaders"
>
  <div
    className="MyMenuResponsiveNav"
  >
    <span
      className="MyMenuTitle"
    >
      Header 2
    </span>
    <button
      className="MyMenuBars fa fa-bars"
      onClick={[Function]}
    />
  </div>
  <div
    className="MyMenuNav"
    displayNav="N"
  >
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 1
    </button>
    <button
      className="MyMenuHeader"
      currentTab="Y"
      onClick={[Function]}
    >
      Header 2
    </button>
    <button
      className="MyMenuHeader"
      currentTab="N"
      onClick={[Function]}
    >
      Header 3
    </button>
  </div>
</div>
`;
